src/autopas/containers/hierarchicalGrids/HierarchicalGrids.h

```diff
@@ -563,77 +538,124 @@ class HierarchicalGrids : public ParticleContainerInterface<Particle> {
+      utils::ExceptionHandler::exception(
+          "HierarchicalGrids: Trying to get the hierarchy level of a particle which is too large to fit in the grid.\n"
+          "Particle radius {}\n"
+          "Upper limit of the largest level {}",
+          particleRad, _hierarchicalGridBorders[level]);
     }
     // Break and return the level
     return level - 1;
   }
 
-  void setHierarchicalGridBorders(std::vector<double> boundaries) {
-
-    _hierarchicalGridBorders = boundaries;
+  void setHierarchicalGridBorders(std::vector<double> boundaries) { _hierarchicalGridBorders = boundaries; }
 
-  }
-
-  std::vector<double> getHierarchicalGridBorders() {
+  std::vector<double> getHierarchicalGridBorders() { return _hierarchicalGridBorders; }
 
-    return _hierarchicalGridBorders;
+ private:
+  /**
+   * Encodes a level number and a cell index as a combined value. The logic behind this is
+   * encodedCellAndLevel = cellIndex + _levelOffset * level
+   * with _levelOffset = 1e15 (see below).
+   *
+   * @note This fails if the number of cells in a level is larger than _levelOffset
+   *
+   * @param level
+   * @param cellIndex
+   * @return size_t Encoded cell and level
+   */
+  size_t encodeCellAndLevel(size_t level, size_t cellIndex) const {
+    // Exception if current level contains too many cells
+    if (cellIndex >= _levelOffset) {
+      utils::ExceptionHandler::exception(
+          "HierarchicalGrids: Trying to encode a cellIndex which is larger than _levelOffset = 1e15,\n"
+          "i.e. the current HG-level contains too many cells.\n"
+          "Current level: {}",
+          level);
+    }
 
+    return cellIndex + _levelOffset * level;
   }
 
+  /**
+   * Decode a encoded cellID and HG level.
+   *
+   * @param encodedCellPlusLevel
+   * @return std::tuple<size_t, size_t> level, cellID
+   */
+  std::tuple<size_t, size_t> decodeCellAndLevel(size_t encodedCellPlusLevel) const {
+    size_t level = encodedCellPlusLevel / _levelOffset;
+    size_t cellID = encodedCellPlusLevel - level * _levelOffset;
 
-  private:
+    return {level, cellID};
+  }
 
   /**
-   * @brief Get the number Of cross-levels of the H-Grid
-   * 
-   * @param levels 
-   * @return unsigned int 
+   * Calculate the number of cross-level interactions for the present number of hierarchy levels
+   *
+   * @return size_t number of cross-level interactions
    */
-  unsigned int getNumberOfCrossLevels(unsigned int levels) {
+  size_t getNumberOfCrossLevelInteractions() const {
+    size_t numberOfCrossLevels = 0;
 
-    unsigned int numberOfCrossLevels = 0;
-    for (unsigned int i = 1; i < levels; i++)
-    {
-      numberOfCrossLevels += i;
+    for (unsigned int largerLevel = 0; largerLevel < _hierarchyLevels.size(); largerLevel++) {
+      for (unsigned int smallerLevel = largerLevel + 1; smallerLevel < _hierarchyLevels.size(); smallerLevel++) {
+        numberOfCrossLevels++;
+      }
     }
-    
+
     return numberOfCrossLevels;
   }
```

Comment:
couldn't this simply return `_crossLevelInteractions.size()`?

Reply:
True